torch/_inductor/ir.py

```diff
@@ -4363,3 +4363,59 @@ def codegen(self, wrapper):
             f"{output_name}_work = dist.all_gather_into_tensor({output_name}, {input_name}, async_op=True,"
             f" group={output_name}_pg)"
         )
+
+class ReduceScatterTensor(ExternKernel):
+    def __init__(
+        self,
+        layout,
+        inputs,
+        constant_args=(),
+    ):
+        super().__init__(None, layout, inputs, constant_args)
+        self.name = V.graph.register_buffer(self)
+
+    def should_allocate(self):
+        return True
+
+    @classmethod
+    def create(cls, x: "TensorBox", reduce_op: str, scatter_dim: int, tag: str, ranks: List[int], group_size: int):
+        x = cls.realize_input(x)
+
+        # is there a difference between literally using x.data.layout below, vs
+        # creating a new one that has the same properties?
+        new_size = x.get_size()
+        new_size[scatter_dim] /= group_size
+        new_layout = FlexibleLayout(x.get_device(), x.get_dtype(), new_size)
+
+        # Collective returns a 'work' object.  But Inductor's scheduler doesn't need to know
+        # about that, and we just pretend for scheduling purposes that the work obj is a 1-elem tensor.
+        # Nobody should consume the output of AllReduce except 'Wait', which we control here.
```

Comment:
nit: For this PR and also for the `all_gather_into_tensor` inductor PR (see [here](https://github.com/pytorch/pytorch/pull/95530/files#diff-cf6ca00beddc32a2a6a2933fb9913b6a2b925ffc3b745488967210e4343134acR4334-R4336)), could we fix the comments to not mention "AllReduce"? This seems like possibly from a copy and paste.

Reply:
you're not supposed to read the comments :P  

(yes, lets  fix!)